recipes/ncurses/all/conanfile.py

```diff
@@ -0,0 +1,186 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+
+
+class NCursesConan(ConanFile):
+    name = "ncurses"
+    description = "The ncurses (new curses) library is a free software emulation of curses in System V Release 4.0 (SVr4), and more"
+    topics = ("conan", "bitcoin", "blockchain", "cryptocurrency", "hash", "p2p")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gnu.org/software/ncurses"
+    license = "X11"
+    exports_sources = "patches/**"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "pkg_config"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_pcre2": [True, False],
+        "with_reentrant": [True, False],
+        "with_widec": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_pcre2": False,
+        "with_reentrant": False,
+        "with_widec": False,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            if self.options.with_widec:
+                raise ConanInvalidConfiguration("with_widec is unsupported for Visual Studio")
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_pcre2:
+            self.requires("pcre2/10.33")
+
+    def build_requirements(self):
+        if self.settings.compiler == "Visual Studio":
+            self.build_requires("getopt-for-visual-studio/20200201")
+            self.build_requires("dirent/1.23.2")
+        if tools.os_info.is_windows and not "CONAN_BASH_PATH" in os.environ and \
+                tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires("msys2/20190524")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("ncurses-{}".format(self.version), self._source_subfolder)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        build = None
+        host = None
+        conf_args = []
+        if self.options.shared:
+            conf_args.extend(["--with-shared", "--with-cxx-shared", "--without-normal"])
```

Comment:
I think you could add cpp option: https://github.com/conan-community/conan-ncurses/blob/release/6.1/conanfile.py#L126

Reply:
I left it out because I thought *the more the better*.
But I've added it again.